src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceService.kt

```diff
@@ -1061,13 +1091,158 @@ class LicenceService(
+    "[INSERT END DATE]",
+    licenceEntity.licenceExpiryDate?.format(dateTimeFormatter)
+      .toString(),
+    true
+  )
+// USE ARD if exits else CRD
+  private fun calculateEndDate(
+    licenceEntity: EntityLicence
+  ): String {
+    val additionalConditionEndDate = licenceEntity.actualReleaseDate
+      ?: licenceEntity.conditionalReleaseDate
+    return additionalConditionEndDate?.plusYears(1)?.format(dateTimeFormatter).toString()
+  }
+
+  private fun hasExistingLicenseContains14B(
+    licenceEntity: EntityLicence
+  ) = licenceEntity.additionalConditions.filter { additionalCondition ->
+    isAdditionalConditionOf14B(additionalCondition.conditionCode!!)
+  }
+
+  private fun hasDateChanged(
+    oldDate: LocalDate?,
+    newDate: LocalDate?
+  ) = nullableDatesDiffer(oldDate, newDate)
+
+  private fun changeTextForLEDChange(
+    updatedLicenceEntity: EntityLicence,
+    licenceEntity: EntityLicence,
+    text: String?,
+    additionalCondition: AdditionalCondition
+  ): String? {
+    var text1 = text
+    if (isLicenseExpiryDateOnOrAfterTwelveMonths(updatedLicenceEntity.licenceExpiryDate)) {
+      // existing license LED before 12 months
+      if (!isLicenseExpiryDateOnOrAfterTwelveMonths(licenceEntity.licenceExpiryDate)) {
+        text1 = replaceExpandedEndDateWithARDOrCRD(updatedLicenceEntity, additionalCondition.conditionText!!)
+      }
+    } else {
+      text1 = replaceExpandedEndDateWithLicenseExpiryDate(additionalCondition.conditionText!!, updatedLicenceEntity)
+    }
+    return text1
+  }
+  private fun changeTextWithEndDate(
+    licenceEntity: EntityLicence,
+    text: String?,
+    updatedLicenceEntity: EntityLicence,
+    additionalCondition: AdditionalCondition
+  ): String? {
+    var text1 = text
+    if (isLicenseExpiryDateOnOrAfterTwelveMonths(licenceEntity.licenceExpiryDate)) {
+      text1 = replaceExpandedEndDateWithARDOrCRD(updatedLicenceEntity, additionalCondition.conditionText!!)
+    }
+    return text1
+  }
+  private fun isARDNull(
+    licenceEntity: EntityLicence
+  ) =
+    (licenceEntity.actualReleaseDate == null)
+
+  private fun sendEmailFor14DateChanges(licenceEntity: EntityLicence, dateChanges: String) {
+    val omuEmail = licenceEntity.prisonCode?.let { omuService.getOmuContactEmail(it)?.email }
+    notifyService.send14BDatesChangedEmail(omuEmail!!, licenceEntity.forename!!, licenceEntity.surname!!, licenceEntity.nomsId!!, dateChanges)
+  }
+
+  private fun getReasonFor14BDateChange(
+    hasLedChange: Boolean,
+    updatedLicenceEntity: EntityLicence,
+    licenceEntity: EntityLicence
+  ): String? {
+    var dateChanges: String? = null
+    if (hasLedChange) {
+      dateChanges = ledDateChangeMsg
+    } else if (hasArdOrCrdChanged(updatedLicenceEntity, licenceEntity)) {
+      dateChanges = crdOrArdChangeMsg
+    }
+    if (hasLedChange && hasArdOrCrdChanged(updatedLicenceEntity, licenceEntity)) {
+      dateChanges = ledAndCrdOrArdChangeMsg
+    }
+
+    return dateChanges
```

Comment:
Should only really ever use mutable references (`var`) when there's no other way around it.

Generally it's best to not worry about having a single exit point - methods should be small enough that having multiple exit points isn't a problem to reason about.

A more idiomatic way of writing this kind of code in kotlin would be to use a `when` ([link](https://kotlinlang.org/docs/control-flow.html#when-expression)): 

```
 private fun getReasonFor14BDateChange(
    hasLedChange: Boolean,
    updatedLicenceEntity: EntityLicence,
    licenceEntity: EntityLicence
  ): String? = when {
    hasLedChange && hasArdOrCrdChanged(updatedLicenceEntity, licenceEntity)) ->  ledAndCrdOrArdChangeMsg
    hasLedChange -> ledDateChangeMsg
    hasArdOrCrdChanged(updatedLicenceEntity, licenceEntity) -> crdOrArdChangeMsg
    else -> null
}
```